lib/AST/UnqualifiedLookup.cpp

```diff
@@ -58,19 +58,34 @@ namespace {
     // lookup.
     if (loc.isInvalid() || decl->getBraces().isInvalid())
       return true;
-    
+
+    SourceManager &SM = decl->getASTContext().SourceMgr;
+
+    // If a code completion happens inside a function body, some lookups may
+    // happen from the 'loc' that is in a different buffer from the 'decl'.
+    // In such cases, look for members because we know 'loc' is inside a
```

Comment:
This comment would be slightly clearer to me by added the words "of the 'decl'" after "look for members".
In other words: "In such cases, look for members of the 'decl' because we know 'loc' is inside a function body in the 'decl'."